crates/configuration/src/to_runtime_configuration.rs

```diff
@@ -171,10 +154,16 @@ fn convert_nullable(nullable: &metadata::Nullable) -> query_engine_metadata::met
 
 fn convert_type(r#type: metadata::Type) -> query_engine_metadata::metadata::Type {
     match r#type {
-        metadata::Type::ScalarType(t) => query_engine_metadata::metadata::Type::ScalarType(t),
         metadata::Type::ArrayType(t) => {
             query_engine_metadata::metadata::Type::ArrayType(Box::new(convert_type(*t)))
-        }
+        },
+        metadata::Type::RangeType(t) => {
+            query_engine_metadata::metadata::Type::RangeType(t)
+        },
+        metadata::Type::StructType(t) => {
+            query_engine_metadata::metadata::Type::StructType(t)
```

Comment:
Weird you don't need to recursively convert the struct members here.

Reply:
Maybe you do 🤷. How can we confirm this?

Reply:
Take a look at the types for both versions of `StructType` I guess. Any idea why we have two versions of `Type`? Seems confusing tbh.

Reply:
The answer is that they are the same type and this function does nothing, deleted it.